Source/missiles.cpp

```diff
@@ -200,22 +200,24 @@ bool MonsterMHit(int pnum, int m, int mindam, int maxdam, int dist, missile_id t
 	if (monster._mmode == MonsterMode::Charge)
 		return false;
 
-	uint8_t mor = monster.mMagicRes;
+	MonsterResists mor = monster.mMagicRes;
 	missile_resistance mir = MissilesData[t].mResist;
 
-	if (((mor & IMMUNE_MAGIC) != 0 && mir == MISR_MAGIC)
-	    || ((mor & IMMUNE_FIRE) != 0 && mir == MISR_FIRE)
-	    || ((mor & IMMUNE_LIGHTNING) != 0 && mir == MISR_LIGHTNING)
-	    || ((mor & IMMUNE_ACID) != 0 && mir == MISR_ACID))
+	if ((mor.isMagicImmune() && mir == MISR_MAGIC) || (mor.isFireImmune() && mir == MISR_FIRE) || (mor.isLightningImmune() && mir == MISR_LIGHTNING) || (mor.isAcidImmune() && mir == MISR_ACID)) {
 		return false;
+	}
 
-	if (((mor & RESIST_MAGIC) != 0 && mir == MISR_MAGIC)
-	    || ((mor & RESIST_FIRE) != 0 && mir == MISR_FIRE)
-	    || ((mor & RESIST_LIGHTNING) != 0 && mir == MISR_LIGHTNING))
-		resist = true;
+	if (mor.isMagicResistant() && mir == MISR_MAGIC) {
+		resist = mor.getMagicResist();
+	} else if (mor.isFireResistant() && mir == MISR_FIRE) {
+		resist = mor.getFireResist();
+	} else if (mor.isLightningResistant() && mir == MISR_LIGHTNING) {
+		resist = mor.getLightningResist();
+	}
 
-	if (gbIsHellfire && t == MIS_HBOLT && (monster.MType->mtype == MT_DIABLO || monster.MType->mtype == MT_BONEDEMN))
-		resist = true;
+	if (gbIsHellfire && t == MIS_HBOLT && (monster.MType->mtype == MT_DIABLO || monster.MType->mtype == MT_BONEDEMN)) {
+		resist = 75;

```

Comment:
```suggestion
		resist = MAX_RESIST;
```

Reply:
I don't think the intent was that Diablo and the Bone Demon have resistance against holy bolt equal to the maximum resistance the player can have. If we don't want a "magic number" here, it should probably be called something else so those two things aren't tied together.

Reply:
It was only a suggestion, I didn't go too deep into the change you are proposing. If doesn't make sense, please ignore my comment.

Reply:
No worries, I was just thinking, I shouldn't have said intent since it's not like they intended for all of this stuff to be modded when they first made Diablo anyway haha - but more from the perspective of it I was modding things, I wouldn't necessarily want those two things linked together, since I might not realize changing `MAX_RESIST` would effect those two things. Regardless, thanks for taking a look 🙂

Reply:
Yes, I totally got your argument and agree. No hard feelings.